### ISSUE TITLE
Phenomenon that type definition does not work when dynamic route of `@hono/zod-openapi` is assigned to `hono/client`.

### ISSUE THREAD

**Issue by tsuki-lab** [honojs/hono#3106]
### What version of Hono are you using?

4.4.12

### What runtime/platform is your app running on?

local

### What steps can reproduce the bug?

Please use the [tsuki-lab/verification-hono-openapi-rpc](https://github.com/tsuki-lab/verification-hono-openapi-rpc/blob/main/src/index.tsx) repository me created as a reproduction environment.

```ts
const app = new OpenAPIHono();

const route = createRoute({
  method: "get",
  path: "/users/{id}", // Use path parameter.
  request: {
    params: z.object({
      id: z.coerce.number().openapi({ description: "userId", example: 1 }),
    }),
  },
  responses: {
    200: {
      content: {
        "application/json": {
          schema: userSchema,
        },
      },
      description: "Get a user",
    },
  },
});

const routes = app.openapi(route, (c) => {
  return c.json([{ id: "1", name: "John" }]);
});

const client = hc<typeof routes>("");
// The type does not match the intended shape
const client: {
  [x: string]: ClientRequest<{
      [Method: `$${Lowercase<string>}`]: Endpoint;
  }>;
}
```

### What is the expected behavior?

`const client` is correctly type-stored.

```ts
const client: {
  user: {
    ":id": {
    // ...
    }
  }
}
```

### What do you see instead?

```ts
const client: {
  [x: string]: ClientRequest<{
      [Method: `$${Lowercase<string>}`]: Endpoint;
  }>;
}
```

**Comment by tsuki-lab** [comment]
I found that this condition reproduces itself when using `z.numer()` and `z.coerce.number()`.
Using `z.string()` is a workaround at this time.
```ts
  // ...
  request: {
    params: z.object({
      id: z.string().openapi({ description: "userId", example: 1 }),
    }),
  },
  // ...
```

However, when using `$get()`, the id passed to param must be converted to a string type each time it is used.

```ts
const r = await client.api.users[":id"].$get({ param: { id: "1" } });
console.log(await r.json());
```

**Comment by yusukebe** [comment]
Hi @tsuki-lab !

This is a bug in `@hono/zod-openapi`. I've fixed it and released a new version `0.14.8`. Try it!

**Comment by tsuki-lab** [comment]
@yusukebe 

Thank you for fixing the bug in `@hono/zod-openapi` and releasing the new version `0.14.8`! I'll try it out right away.
